packages/wallet/backend/migrations/20240215061440_add_delete_flag_to_transactions_status_check_constraint.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function (knex) {
+  return knex.schema.alterTable('transactions', async (table) => {
+    await knex.raw(
+      [
+        `ALTER TABLE "transactions" DROP CONSTRAINT IF EXISTS "transactions_status_check";`,
+        `ALTER TABLE "transactions" ADD CONSTRAINT "transactions_status_check" CHECK ("status" = ANY (ARRAY['PENDING'::text, 'COMPLETED'::text, 'REJECTED'::text, 'EXPIRED'::text, 'DELETED'::text]))`
+      ].join('\n')
+    )
+
+    await knex.raw(
+      [
+        `UPDATE transactions`,
```

Comment:
a similar query should be added in `down` to restore the data.

Reply:
@beniaminmunteanu @Tymmmy @raducristianpopa 
I think it is not the best to add a new status in this case as the previous one is overwritten.
Keeping the status as it is and adding a separate field to indicate that transactions is deleted/inactive allows to have the correct transactions status for history and can be easily restored.

Reply:
Yes, I agree, we need a new field here to indicate that it is an inactive PP

Reply:
good point.